xidmap/trie.go

```diff
@@ -128,3 +134,33 @@ func (t *Trie) put(offset uint32, key string, uid uint64) uint32 {
 	}
 	return offset
 }
+
+func (t *Trie) iterate(offset uint32, prefix string, fn iterFn) error {
+	if offset == 0 {
+		return nil
+	}
+
+	n := t.getNode(offset)
+	if n == nil {
+		return nil
+	}
+
+	if err := t.iterate(n.left, prefix, fn); err != nil {
+		return err
+	}
+
+	if n.uid != 0 {
+		if err := fn(prefix+string(n.r), n.uid); err != nil {
+			return err
+		}
+	}
+	if err := t.iterate(n.mid, prefix+string(n.r), fn); err != nil {
```

Comment:
hey @NamanJain8 , why do we iterate the n.mid with `prefix+string(n.r)` while the left and right are iterated with `prefix`?